src/wamr/wamr.cc

```diff
@@ -167,7 +166,7 @@ std::unique_ptr<WasmVm> Wamr::clone() {
 
   WasmByteVec stripped;
   clone->module_ =
-      wasm_module_new(store_.get(), getStrippedSource(&stripped) ? stripped.get() : source_.get());
+      wasm_module_new(clone->store_.get(), getStrippedSource(&stripped) ? stripped.get() : source_.get());
   return clone;
 }
 
```

Comment:
So we don't need to change the link function implemention?

Reply:
yes. Envoy uses that function to inject lots of hosted functions into WASM world.  Can not ignore them at all.